portal-api/src/v1/gef/services/clone/clone-deal.service.ts

```diff
@@ -0,0 +1,86 @@
+import { ObjectId } from 'mongodb';
+import {
+  AuditDetails,
+  Bank,
+  PortalSessionUser,
+  GefDeal,
+  MONGO_DB_COLLECTIONS,
+  InvalidDealIdError,
+  InvalidParameterError,
+  DealNotFoundError,
+} from '@ukef/dtfs2-common';
+import { mongoDbClient } from '../../../../drivers/db-client';
+import { getLatestEligibilityCriteria } from '../../controllers/eligibilityCriteria.controller';
+import { generateClonedDeal } from './helpers/generate-cloned-deal';
+
+type CloneDealParams = {
+  dealId: string | ObjectId;
+  bankInternalRefName: string;
+  additionalRefName: string;
+  maker: PortalSessionUser;
+  userId: string | ObjectId;
+  bank: Bank;
+  auditDetails: AuditDetails;
+};
+
+/**
+ * Creates a copy of existing GEF deal.
+ * Inserts a clone into the database as the latest version.
+ * Updates/removes certain values.
+ *
+ * @param params 
+ * @param params.dealId - the deal ID
+ * @param params.bankInternalRefName - the new deals internal ref name
+ * @param params.additionalRefName - new new deals additional ref name
+ * @param params.maker - the user making the clone
+ * @param params.userId - the makers id
+ * @param params.bank - the users bank
+ * @param params.auditDetails - the users audit details
+
+ * @returns the inserted deal id
+ */
+export const cloneDealToLatestVersion = async ({
```

Comment:
Is this not being used anywhere correct right now?

Reply:
This is used in dtfs2\portal-api\src\v1\gef\controllers\clone-gef-deal.controller.js

Reply:
I did not pull. Thanks!